gemd/entity/base_entity.py

```diff
@@ -70,3 +70,18 @@ def add_uid(self, scope, uid):
 
         """
         self.uids[scope] = uid
+
+    # Note that this could violate transitivity -- Link(scope1) == obj == Link(scope2)
+    def __eq__(self, other):
+        from gemd.entity.link_by_uid import LinkByUID
+        if isinstance(other, LinkByUID):
+            return self.uids.get(other.scope) == other.id
+        else:
+            result = super().__eq__(other)
+            return result
+
+    # Note the hash function checks if objects are identical, as opposed to the equals method,
+    # which checks if fields are equal.  This is because BaseEntities are fundamentally
+    # mutable objects.
+    def __hash__(self):
```

Comment:
What's the purpose of this definition?  Doesn't it automatically inherit from its parent?

Reply:
It actually does not; you need to explicitly define a __hash__ function every time you define an __eq__.  I went into this expecting ^^ would have been correct.

Reply:
https://docs.python.org/3/reference/datamodel.html#object.__hash__

Reply:
That might be worth mentioning in the inline comments